etc/systemd/system/zfs-volume@.service.in

```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=ZFS zvol %I ready
+Before=zfs-volumes.target
+DefaultDependencies=no
+
+[Service]
+ExecStart=sh -c 'timeout=30 ; while [ $timeout -gt 0 ] ; do [ -e "/dev/zvol/$0" ] && exit 0 ; sleep 1 ; timeout=$((timeout-1)); done ; echo "zvol $0 unavailable"' %I
```

Comment:
The man page (in Debian and [Ubuntu](https://manpages.ubuntu.com/manpages/cosmic/man5/systemd.service.5.html)), says that "It is thus safe to use just the executable name in case of executables located in any of the "standard" directories".  It does recommend absolute paths to "avoid ambiguity". Could you confirm that this is actually broken in Ubuntu? If there's even one example where this could be a problem, I think it's worth adjusting.

Reply:
Omfg. Ubuntu 12.04, 18.10, 19.04 allow non-absolute paths while 14.04, 16.04, 18.04 requires absolute paths (they fail with *Executable path is not absolute, ignoring: sh -c*). WTF are they doing...

Short: yes, ExecStart must be absolute here

Reply:
Thanks for testing this @c0d3z3r0. Sorry I missed your comment @aerusso. This is indeed madness...

Reply:
@c0d3z3r0 Strange! Thanks for testing, I've updated this to use `/bin/sh`.